Add "How Via 3 works" to README (#61)
hypothesis/via: README.md

`make help` to see all the commands that're available for running the tests,
linting, code formatting, etc.

Updating the PDF viewer
-----------------------

Via 3 serves PDFs using [PDF.js](https://mozilla.github.io/pdf.js/). PDF.js is
vendored into the source tree and the viewer HTML is patched to load the Hypothesis
client. To update the PDF viewer, run `tools/update-pdfjs`.

How Via 3 works
---------------

> Via 3 allows user to annotate arbitrary web pages or PDF files by proxying the

Author: ```suggestion
Via 3 allows users to annotate arbitrary web pages or PDF files by proxying the
```

page or file and injecting the Hypothesis client. Users go to
<https://via3.hypothes.is/> and paste in a PDF or HTML URL (or visit
`https://via3.hypothes.is/<SOME_URL>` directly) and Via 3 responds with an
annotatable version.

### Via 3's architecture

Via 3 is composed of four separable components:

1. A **top-level component** that responds to requests to the top-level
   `/<THIRD_PARTY_URL>` endpoint by deciding whether the URL is a PDF file or
   not and redirecting the browser to either the PDF viewer component or the
   HTML proxying component accordingly.

   This component is implemented in Python / Pyramid.

>    The Pyramid app sends an HTTP GET request to the third-party URL but only

Author: ```suggestion
   The Pyramid app sends a GET request to the third-party URL but only
```
